src/store/flow/resource.ts

```diff
@@ -28,48 +27,55 @@ import {IFlowsState} from '@/store/flow/index'
 import {IRootState} from '@/store'
 
 export const getters: GetterTree<IFlowsState, IRootState> = {
-  resourcesByUuid: ({resources}) => keyBy(resources, 'uuid'),
+  resourcesByUuidOnActiveFlow: (_state, getters) => keyBy(getters.activeFlow.resources, 'uuid'),
 
-  resourceUuidsOnActiveFlow: (state, getters) => filter(map(getters.activeFlow.blocks, (block) => block.config.prompt)),
-
-  resourcesOnActiveFlow: (state, getters): IResource[] => filter(
-    state.resources,
-    (resource) => includes(getters.resourceUuidsOnActiveFlow, resource.uuid)
-  ),
+  resourceUuidsOnActiveFlow: (_state, getters) => filter(map(getters.activeFlow.resources, (res) => res.uuid)),
 }
 
-export const mutations: MutationTree<IFlowsState> = {
-  resource_add({resources}, {resource}: { resource: IResource }) {
-    resources.push(resource)
+export const mutations: MutationTree<IFlowsState> = {}
+
+export const actions: ActionTree<IFlowsState, IRootState> = {
+  resource_add({getters}, {resource}: {resource: IResource}) {
```

Comment:
What's the reason to move to an action here? I see you are catching a null resource, what was causing this situation?

Are there consequences to this now being async?

Reply:
The main reason to use `actions` are to access at the `getters.activeFlow` which avoids us to spread `flowId` as param everywhere.

After seeing your comment below, I brought changes on these codes. Pls review again.